controllers/cluster/rbac.go

```diff
@@ -41,7 +41,16 @@ func createClusterRBAC(obj pipeline.Object, data *pipeline.Context) pipeline.Res
 		}},
 	}
 	for _, item := range []client.Object{serviceAccount, role, roleBinding} {
-		if err := data.Client.Create(context.TODO(), item); err != nil && !errors.IsAlreadyExists(err) {
+		found := item.DeepCopyObject().(client.Object)
+		err := data.Client.Get(context.TODO(), client.ObjectKeyFromObject(item), found)
+		if errors.IsNotFound(err) {
+			if err := data.Client.Create(context.TODO(), item); err != nil && !errors.IsAlreadyExists(err) {
+				return pipeline.Result{Err: err}
+			}
+		} else if err != nil {
+			return pipeline.Result{Err: err}
+		}
```

Comment:
I suggest to first check if err is not nil, an only then run the errors.IsNotFound check.